sdk/tests/core/test_az_json.c

```diff
@@ -3068,16 +3102,24 @@ static void test_az_json_string_unescape(void** state)
     destination_span._internal.size = final_size;
 
     assert_int_equal(AZ_OK, result);
-    assert_int_equal(expected._internal.size, final_size);
-    assert_string_equal(expected._internal.ptr, destination);
     assert_true(az_span_is_content_equal(expected, destination_span));
   }
 
-  // malformed escapes
+  // insufficient space
   {
-    az_span original = AZ_SPAN_FROM_STR("abcd\\");
-    az_span expected = AZ_SPAN_FROM_STR("abcd\\");
+    az_span json = AZ_SPAN_FROM_STR(" { \"name\": \"some value string\" , \"code\" : 123456 } ");
+    char destination[5] = { 0 };
+    int final_size;
+
+    az_result result = az_json_string_unescape(json, destination, 4, &final_size);
 
+    assert_int_equal(AZ_ERROR_NOT_ENOUGH_SPACE, result);
+  }
+
+  // magic test
+  {
+    az_span original = AZ_SPAN_FROM_STR("A\\\"\"Z");
```

Comment:
Ideally, we should disallow such inputs since having an unescaped embedded quote within a JSON string token is invalid.

Rather than this succeeding, it should return az_result unexpected char.

That said, we can resolve this in a follow-up issue.

Reply:
it was one of the follow up issues we discussed yesterday